### PR TITLE
Feature/bac 881 aansluiten catalogus op frontend

### PR DESCRIPTION
**Pull request by aditudorache**
The dcatd catalog detail can be reviewed now.

### CODE REVIEW

File: modules/detail/components/detail/templates/dcatd/styles/_fonts.scss [hunk 1]
```diff
@@ -0,0 +1,32 @@
+%font {
```
Comment (boris-arkenaar) [inline modules/detail/components/detail/templates/dcatd/styles/_fonts.scss]: Dit zijn ook allemaal standaard (shared) dingen.

File: modules/detail/components/option-label/option-label.filter.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+(function () {
+    'use strict';
+
+    angular
+        .module('dpDetail')
+        .filter('optionLabel', optionLabelFilter);
+
+    function optionLabelFilter () {
+        return function (input, list, namespace) {
+            // returns the label of an option from a option list
```
Comment (boris-arkenaar) [inline modules/detail/components/option-label/option-label.filter.js]: `an option`

File: modules/detail/components/option-label/option-label.filter.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+            // the list array elements contain at list id and label properties
```
Comment (boris-arkenaar) [inline modules/detail/components/option-label/option-label.filter.js]: `at` => `a`?

File: modules/detail/components/option-label/option-label.filter.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+            let prefix = '';
+            if (angular.isDefined(namespace)) prefix = `${namespace}:`;
```
Comment (boris-arkenaar) [inline modules/detail/components/option-label/option-label.filter.js]: `const prefix = angular.isDefined(namespace) ? `${namespace}:` : '';

File: modules/detail/components/option-label/option-label.filter.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+            const index = list.findIndex((item) => (prefix + item.id) === input);
+            if (index > -1) {
```
Comment (boris-arkenaar) [inline modules/detail/components/option-label/option-label.filter.js]: `return index > -1 ? list[index].label : input;`

File: modules/detail/services/data-formatter/data-formatter.factory.js [hunk 6]
```diff
@@ -5,7 +5,9 @@
         .module('dpDetail')
         .factory('dataFormatter', dataFormatterFactory);
 
-    function dataFormatterFactory () {
+    dataFormatterFactory.$inject = ['store'];
+
+    function dataFormatterFactory (store) {
```
Comment (boris-arkenaar) [inline modules/detail/services/data-formatter/data-formatter.factory.js]: Bad practice; services weten niets van de store af.

File: modules/detail/services/data-formatter/data-formatter.factory.js [hunk 7]
```diff
@@ -43,5 +47,24 @@
                 return result;
             }, formattedData);
         }
+
+        function formatCatalogData (data) {
+            const state = store.getState();
+            const resourceTypes = state.catalogFilters.resourceTypes;
+            if (!resourceTypes) return {};
```
Comment (boris-arkenaar) [inline modules/detail/services/data-formatter/data-formatter.factory.js]:
```
if (!resourceTypes) {
  return {};
}
```

File: modules/data-selection/services/api/data-selection-api-dcatd.factory.js [hunk 8]
```diff
@@ -0,0 +1,178 @@
...
+                // optional format filter
+                searchParams[searchParamFormat] = queryFormat;
+            }
+
+            if (queryOwner !== '') {
+                // optional owner filter
+                searchParams[searchParamOwner] = queryOwner;
+            }
+
+            if (Object.keys(catalogFilters).length === 0 && catalogFilters.constructor === Object) {
```
[the remark this answers is not in the document]
Comment (aditudorache) [reply modules/data-selection/services/api/data-selection-api-dcatd.factory.js]: ok.

File: modules/data-selection/services/api/data-selection-api-dcatd.factory.js [hunk 9]
```diff
@@ -0,0 +1,178 @@
...
+        }
+
+        function getQueryOwner (filters) {
+            return Object.keys(filters).reduce((queryString, key) => {
+                if (key !== 'owner') return queryString;
+                return queryString + `eq=${filters[key]}`;
+            }, '');
+        }
+
+        function formatFilters (searchParams, filtersConfig, catalogFilters) {
```
[the remark this answers is not in the document]
Comment (aditudorache) [reply modules/data-selection/services/api/data-selection-api-dcatd.factory.js]: removed

File: src/catalog/ducks/data-selection/data-selection-catalog.js [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+      options.push({
+        id: index === -1 ? propertyType.enum[i] : propertyType.enum[i].substring(index + 1),
+        label: propertyType.enumNames[i]
+      });
+    }
+  );
+  return options;
+}
+
+function getCatalogFilters(data) {
```
[the remark this answers is not in the document]
Comment (aditudorache) [reply src/catalog/ducks/data-selection/data-selection-catalog.js]: Moved to the service.

File: modules/data-selection/services/api/data-selection-api-dcatd.factory.js [hunk 11]
```diff
@@ -0,0 +1,178 @@
...
+                    return {
+                        id: option.id,
+                        label: option.label,
+                        count: 1
+                    };
+                })
+            };
+
+            return Object.keys(filters).reduce((filterString, key) => {
+                return filters;
```
[the remark this answers is not in the document]
Comment (boris-arkenaar) [reply modules/data-selection/services/api/data-selection-api-dcatd.factory.js]: Huh, it already was `return filters`. What's the use of this piece of code?

File: modules/data-selection/services/api/data-selection-api-dcatd.factory.test.response.json [hunk 12]
```diff
@@ -0,0 +1,89 @@
...
+            "dct:title": "Evenementen in Amsterdam"
+        },
+        {
+            "@id": "ams-dcatd:d7a4c93c-0d7f-4d39-82d4-5f50eaffa624",
+            "dct:description": "<p>Alle attracties in Amsterdam en omgeving...</p>",
+            "dct:identifier": "d7a4c93c-0d7f-4d39-82d4-5f50eaffa624",
+            "dct:title": "Attracties"
+        }
+    ]
+}
```
[the remark this answers is not in the document]
Comment (boris-arkenaar) [reply modules/data-selection/services/api/data-selection-api-dcatd.factory.test.response.json]: Yup. A file should end with a line feed.

File: modules/data-selection/services/api/data-selection-api-dcatd.factory.js [hunk 13]
```diff
@@ -65,98 +65,61 @@
...
-            return Object.keys(filters).reduce((queryString, key) => {
-                if (key !== 'owner') return queryString;
-                return queryString + `eq=${filters[key]}`;
-            }, '');
-        }
-
-        function formatFilters (searchParams, filtersConfig, catalogFilters) {
-            var filters = {
+        function formatFilters (filtersConfig, catalogFilters) {
+            const filters = {
```
Comment (boris-arkenaar) [inline modules/data-selection/services/api/data-selection-api-dcatd.factory.js]: `return {`

File: modules/data-selection/services/api/data-selection-api-dcatd.factory.js (hunk 11 above)
[the remark this answers is not in the document]
Comment (boris-arkenaar) [reply modules/data-selection/services/api/data-selection-api-dcatd.factory.js]: I see what you mean!

File: modules/detail/components/detail/templates/dcatd/_datasets.scss [hunk 14]
```diff
@@ -114,3 +124,8 @@
     }
   }
 }
+
+
```
Comment (boris-arkenaar) [inline modules/detail/components/detail/templates/dcatd/_datasets.scss]: One empty line is enough

File: modules/detail/components/detail/templates/dcatd/partials/resources.html [hunk 15]
```diff
@@ -20,12 +20,12 @@
...
                                 <span ng-if="!resource['dct:format']"> ? </span>
                             </span>
-                            <span>{{resource['dct:description']}}</span>
+                            <span>{{resource['dct:description'] ? resource['dct:description'] : resource['dcat:accessURL'] }}</span>
                         </div>
                     </div>
                     <div class="resources-item__right">
                         <div class="resources-item__modified">
-                            gewijzigd op {{resource['dct:modified'] | date:'dd MMMM yyyy' }}
+                            <span ng-if="resource['dct:modified']">gewijzigd op {{resource['dct:modified'] | date:'dd MMMM yyyy' }}</span>&nbsp;
```
Comment (boris-arkenaar) [inline modules/detail/components/detail/templates/dcatd/partials/resources.html]: Is die `&nbsp;` nodig? Het zit in een `div`...

File: modules/detail/components/option-label/option-label.filter.js [hunk 16]
```diff
@@ -7,16 +7,11 @@
...
-            if (index > -1) {
-                return list[index].label;
-            } else {
-                return input;
-            }
+            // returns the label of an option from an option list
+            // the list array elements contain at least the id and label properties
+            const prefix = angular.isDefined(namespace) ? `${namespace}:` : '';
+            const index = list && list.findIndex((item) => (prefix + item.id) === input);
+            return index > -1 ? list[index].label : input;
```
Comment (boris-arkenaar) [inline modules/detail/components/option-label/option-label.filter.js]: Nice and clean!

File: modules/detail/services/data-formatter/data-formatter.factory.js [hunk 17]
```diff
@@ -51,20 +51,31 @@
...
 
-            data.resources = resourceTypes.map((item, index) => {
-                return {
-                    type: item.id,
-                    rows: resources.filter((row) => row['ams:resourceType'] === item.id) // .map((r) => {r['dct:modified'] = '2018-01-07'; return r;})
-                };
-            }).filter(resource => resource.rows.length > 0);
+            const formattedData = {
+                _display: data['dct:title'],
+                resources: resourceTypes.map((item, index) => {
```
Comment (boris-arkenaar) [inline modules/detail/services/data-formatter/data-formatter.factory.js]: `index` is not being used.

File: modules/detail/services/data-formatter/data-formatter.factory.js [hunk 18]
```diff
@@ -51,20 +51,31 @@
...
+                    return {
+                        type: item.id,
+                        rows: resources.filter((row) => row['ams:resourceType'] === item.id)
+                    };
+                }).filter(resource => resource.rows.length),
+                editDatasetUrl: `dcatd_admin/datasets/${data['dct:identifier']}`,
+                canEditDataset: state.user.scopes.includes('CAT/W')
+            };
+
+            return Object.keys(data).reduce((result, key) => {
```
Comment (boris-arkenaar) [inline modules/detail/services/data-formatter/data-formatter.factory.js]: Very nice! Another possible way to do it:
```
return Object
  .keys(data)
  .filter((key) => key !== 'dcat:distribution')
  .reduce((acc, key) => ({
    ...acc,
    [key]: data[key]
  }), formattedData);
```

File: modules/page/components/homepage/catalogus-themes/_catalogus-themes.scss [hunk 19]
```diff
@@ -92,3 +92,5 @@ $base-padding: calc(#{$base-whitespace} * 2);
 .c-catalogus-theme__icon--educatie-jeugd-diversiteit {
   @include icon-button-text-right("educatie-jeugd-diversiteit", "", "red");
 }
+
+
```
Comment (boris-arkenaar) [inline modules/page/components/homepage/catalogus-themes/_catalogus-themes.scss]: A bit too much empty lines? You got your linter going?

File: src/shared/styles/config/mixins/_icons-thema.scss [hunk 20]
```diff
@@ -0,0 +1,96 @@
+
+%svg-common-thema {
```
Comment (boris-arkenaar) [inline src/shared/styles/config/mixins/_icons-thema.scss]: Is this file not going to be generated by the grunt build task?

File: src/catalog/services/catalog-filters.js [hunk 21]
```diff
@@ -1,6 +1,42 @@
...
+    temporalUnits: getOptions(dcatDocProperties['ams:temporalUnit']),
+    accrualPeriodicities: getOptions(dcatDocProperties['dct:accrualPeriodicity']),
+    languages: getOptions(dcatDocProperties['dct:language'])
+  };
+
+  return catalogFilters;
+}
+
+export function fetchFilters() {
   return getByUrl(`${apiUrl}dcatd/openapi`);
```
Comment (boris-arkenaar) [inline src/catalog/services/catalog-filters.js]: Ik zou `getCatalogFilters` hier gelijk toepassen. En dan hoeven zowel `getCatalogFilters` als `getOptions` niet meer geëxporteerd te worden.

File: modules/detail/services/data-formatter/data-formatter.factory.js (hunk 6 above)
Comment (boris-arkenaar) [reply modules/detail/services/data-formatter/data-formatter.factory.js]: Pass it along from where this service is being called; `data = dataFormatter.formatData(data, subject, vm.catalogFilters);` => https://github.com/Amsterdam/atlas/pull/1112/files#diff-cfead18cfb173664f61e596caca21326R101

File: modules/detail/components/detail/templates/dcatd/_datasets.scss (hunk 14 above)
Comment (aditudorache) [reply modules/detail/components/detail/templates/dcatd/_datasets.scss]: Done

File: modules/detail/components/detail/templates/dcatd/partials/resources.html (hunk 15 above)
Comment (aditudorache) [reply modules/detail/components/detail/templates/dcatd/partials/resources.html]: Is wel nodig om de verhouding/height te behouden when the div is empty.

File: modules/detail/components/option-label/option-label.filter.js (hunk 16 above)
Comment (aditudorache) [reply modules/detail/components/option-label/option-label.filter.js]: Thanks

File: modules/detail/services/data-formatter/data-formatter.factory.js (hunk 17 above)
Comment (aditudorache) [reply modules/detail/services/data-formatter/data-formatter.factory.js]: removed

File: modules/detail/services/data-formatter/data-formatter.factory.js (hunk 18 above)
Comment (aditudorache) [reply modules/detail/services/data-formatter/data-formatter.factory.js]: Thanks!

File: modules/page/components/homepage/catalogus-themes/_catalogus-themes.scss (hunk 19 above)
Comment (aditudorache) [reply modules/page/components/homepage/catalogus-themes/_catalogus-themes.scss]: It seems the test-lint doesnt find all problems, I'll look into it

File: src/shared/styles/config/mixins/_icons-thema.scss (hunk 20 above)
Comment (aditudorache) [reply src/shared/styles/config/mixins/_icons-thema.scss]: You are right, removed from git.

File: src/catalog/services/catalog-filters.js (hunk 21 above)
Comment (aditudorache) [reply src/catalog/services/catalog-filters.js]: done

File: modules/.eslintrc.js [hunk 22]
```diff
@@ -146,7 +146,8 @@ module.exports = {
         "no-invalid-this": "error",             // disallow this keywords outside of classes or class-like objects
         "require-yield": "error",               // require generator functions to contain yield
         "jasmine/no-focused-tests": "error",    // disallow fdescribe, fit
-        "jasmine/no-disabled-tests": "error"    // disallow xdescribe, xit
+        "jasmine/no-disabled-tests": "error",    // disallow xdescribe, xit
```
Comment (boris-arkenaar) [inline modules/.eslintrc.js]: Correct the position of the comment

File: modules/data-selection/services/api/data-selection-api-dcatd.factory.test.response.json [hunk 23]
```diff
@@ -1,28 +1,27 @@
...
-            "dct:title": "Evenementen in Amsterdam"
-        },
-        {
-            "@id": "ams-dcatd:d7a4c93c-0d7f-4d39-82d4-5f50eaffa624",
-            "dct:description": "<p>Alle attracties in Amsterdam en omgeving...</p>",
-            "dct:identifier": "d7a4c93c-0d7f-4d39-82d4-5f50eaffa624",
-            "dct:title": "Attracties"
-        }
-    ]
+  "dcat:dataset": [{
```
Comment (boris-arkenaar) [inline modules/data-selection/services/api/data-selection-api-dcatd.factory.test.response.json]: Doe die `{` dan ook even op zijn eigen regel.

File: src/catalog/services/catalog-filters.js [hunk 24]
```diff
@@ -37,6 +39,6 @@ export function getCatalogFilters(data) {
   return catalogFilters;
 }
 
-export function fetchFilters() {
-  return getByUrl(`${apiUrl}dcatd/openapi`);
+export default function fetchFilters() {
+  return Promise.resolve(getByUrl(`${apiUrl}dcatd/openapi`).then((data) => getCatalogFilters(data)));
```
Comment (boris-arkenaar) [inline src/catalog/services/catalog-filters.js]: This is fine right?
Look for example at: https://github.com/Amsterdam/atlas/blob/develop/src/shared/services/meetbout/meetbout.js

But is the `Promise.resolve` necessary?
Can't you just do
```
return getByUrl(`${apiUrl}dcatd/openapi`)
  .then((data) => getCatalogFilters(data));
```

File: modules/detail/components/detail/templates/dcatd/partials/resources.html (hunk 15 above)
Comment (aditudorache) [reply modules/detail/components/detail/templates/dcatd/partials/resources.html]: oke

File: modules/.eslintrc.js (hunk 22 above)
Comment (aditudorache) [reply modules/.eslintrc.js]: thanks

File: modules/data-selection/services/api/data-selection-api-dcatd.factory.test.response.json (hunk 23 above)
Comment (aditudorache) [reply modules/data-selection/services/api/data-selection-api-dcatd.factory.test.response.json]: ok

File: src/catalog/services/catalog-filters.js (hunk 24 above)
Comment (aditudorache) [reply src/catalog/services/catalog-filters.js]: Right! Thanks!

**Pull request merged**